src/main/java/appeng/util/item/AEItemStackRegistry.java

```diff
@@ -31,16 +31,33 @@
 
 import net.minecraft.item.ItemStack;
 
+import appeng.util.Platform;
+
 
 public final class AEItemStackRegistry
 {
-	private static final WeakHashMap<AESharedItemStack, WeakReference<AESharedItemStack>> REGISTRY = new WeakHashMap<>();
+	private static final WeakHashMap<AESharedItemStack, WeakReference<AESharedItemStack>> SERVER_REGISTRY = new WeakHashMap<>();
+	private static final WeakHashMap<AESharedItemStack, WeakReference<AESharedItemStack>> CLIENT_REGISTRY = new WeakHashMap<>();
+
+	private static long nextStackId = 0l;
 
 	private AEItemStackRegistry()
 	{
 	}
 
-	static synchronized AESharedItemStack getRegisteredStack( final @Nonnull ItemStack itemStack )
+	static private WeakHashMap<AESharedItemStack, WeakReference<AESharedItemStack>> registry()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)